2020-02-18 16:18:52.029  INFO 26052 --- [main] i.d.d.DeveloperServiceApplication        : Starting DeveloperServiceApplication on Stevens-MacBook-Pro.local with PID 26052 (/Users/stevenmcdonald/Documents/developer-servicev-1/build/classes/java/main started by stevenmcdonald in /Users/stevenmcdonald/Documents/developer-servicev-1)
2020-02-18 16:18:52.032  INFO 26052 --- [main] i.d.d.DeveloperServiceApplication        : No active profile set, falling back to default profiles: default
2020-02-18 16:18:52.559  INFO 26052 --- [main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data Reactive MongoDB repositories in DEFAULT mode.
2020-02-18 16:18:52.573  INFO 26052 --- [main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 10ms. Found 0 Reactive MongoDB repository interfaces.
2020-02-18 16:18:52.576  INFO 26052 --- [main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data MongoDB repositories in DEFAULT mode.
2020-02-18 16:18:52.577  INFO 26052 --- [main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 0ms. Found 0 MongoDB repository interfaces.
2020-02-18 16:18:52.984  INFO 26052 --- [main] org.mongodb.driver.cluster               : Cluster created with settings {hosts=[localhost:27017], mode=SINGLE, requiredClusterType=UNKNOWN, serverSelectionTimeout='30000 ms', maxWaitQueueSize=500}
2020-02-18 16:18:53.060  INFO 26052 --- [cluster-ClusterId{value='5e4c62cc5dd0cd2db7f8dc2b', description='null'}-localhost:27017] org.mongodb.driver.cluster               : Exception in monitor thread while connecting to server localhost:27017

com.mongodb.MongoSocketOpenException: Exception opening socket
	at com.mongodb.connection.netty.NettyStream$OpenChannelFutureListener.operationComplete(NettyStream.java:410) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.connection.netty.NettyStream$OpenChannelFutureListener.operationComplete(NettyStream.java:385) ~[mongodb-driver-core-3.11.2.jar:na]
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:577) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:570) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:549) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:490) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:615) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:608) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:117) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.fulfillConnectPromise(AbstractNioChannel.java:321) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:337) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:702) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_241]
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017
Caused by: java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method) ~[na:1.8.0_241]
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717) ~[na:1.8.0_241]
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:330) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:702) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_241]

2020-02-18 16:18:53.184  INFO 26052 --- [main] o.s.b.a.e.web.EndpointLinksResolver      : Exposing 2 endpoint(s) beneath base path '/actuator'
2020-02-18 16:18:53.442  INFO 26052 --- [main] o.s.b.web.embedded.netty.NettyWebServer  : Netty started on port(s): 8080
2020-02-18 16:18:53.444  INFO 26052 --- [main] i.d.d.DeveloperServiceApplication        : Started DeveloperServiceApplication in 1.682 seconds (JVM running for 2.385)
2020-02-18 16:18:53.719  INFO 26052 --- [RMI TCP Connection(2)-127.0.0.1] org.mongodb.driver.cluster               : No server chosen by com.mongodb.async.client.ClientSessionHelper$1@457bad92 from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
2020-02-18 16:19:23.733  INFO 26052 --- [cluster-5e4c62cc5dd0cd2db7f8dc2b] org.mongodb.driver.cluster               : No server chosen by ReadPreferenceServerSelector{readPreference=primary} from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
2020-02-18 16:19:30.381  WARN 26052 --- [cluster-5e4c62cc5dd0cd2db7f8dc2b] o.s.b.a.m.MongoReactiveHealthIndicator   : Mongo health check failed

org.springframework.data.mongodb.UncategorizedMongoDbException: Shutdown in progress; nested exception is com.mongodb.MongoClientException: Shutdown in progress
	at org.springframework.data.mongodb.core.MongoExceptionTranslator.translateExceptionIfPossible(MongoExceptionTranslator.java:138) ~[spring-data-mongodb-2.2.4.RELEASE.jar:2.2.4.RELEASE]
	at org.springframework.data.mongodb.core.ReactiveMongoTemplate.potentiallyConvertRuntimeException(ReactiveMongoTemplate.java:2777) ~[spring-data-mongodb-2.2.4.RELEASE.jar:2.2.4.RELEASE]
	at org.springframework.data.mongodb.core.ReactiveMongoTemplate.lambda$translateException$87(ReactiveMongoTemplate.java:2760) ~[spring-data-mongodb-2.2.4.RELEASE.jar:2.2.4.RELEASE]
	at reactor.core.publisher.Flux.lambda$onErrorMap$28(Flux.java:6430) ~[reactor-core-3.3.2.RELEASE.jar:3.3.2.RELEASE]
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:88) ~[reactor-core-3.3.2.RELEASE.jar:3.3.2.RELEASE]
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyInner.onError(MonoFlatMapMany.java:247) ~[reactor-core-3.3.2.RELEASE.jar:3.3.2.RELEASE]
	at com.mongodb.reactivestreams.client.internal.SingleResultObservableToPublisher$1.onError(SingleResultObservableToPublisher.java:50) ~[mongodb-driver-reactivestreams-1.12.0.jar:na]
	at com.mongodb.reactivestreams.client.internal.ObservableToPublisher$1.onError(ObservableToPublisher.java:71) ~[mongodb-driver-reactivestreams-1.12.0.jar:na]
	at com.mongodb.async.client.AbstractSubscription.onError(AbstractSubscription.java:129) ~[mongodb-driver-async-3.11.2.jar:na]
	at com.mongodb.async.client.SingleResultCallbackSubscription$1.onResult(SingleResultCallbackSubscription.java:42) ~[mongodb-driver-async-3.11.2.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.async.client.OperationExecutorImpl$1$1$1.onResult(OperationExecutorImpl.java:94) ~[mongodb-driver-async-3.11.2.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.operation.OperationHelper$ReferenceCountedReleasingWrappedCallback.onResult(OperationHelper.java:412) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.operation.CommandOperationHelper$8.call(CommandOperationHelper.java:413) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.operation.OperationHelper$AsyncCallableWithConnectionAndSourceCallback.onResult(OperationHelper.java:629) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.operation.OperationHelper$AsyncCallableWithConnectionAndSourceCallback.onResult(OperationHelper.java:619) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.binding.AsyncClusterBinding$1.onResult(AsyncClusterBinding.java:111) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.binding.AsyncClusterBinding$1.onResult(AsyncClusterBinding.java:107) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.internal.connection.BaseCluster$ServerSelectionRequest.onResult(BaseCluster.java:440) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.internal.connection.BaseCluster$WaitQueueHandler.run(BaseCluster.java:512) ~[mongodb-driver-core-3.11.2.jar:na]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_241]
Caused by: com.mongodb.MongoClientException: Shutdown in progress
	... 2 common frames omitted

2020-02-18 16:21:18.162  INFO 26216 --- [main] i.d.d.DeveloperServiceApplication        : Starting DeveloperServiceApplication on Stevens-MacBook-Pro.local with PID 26216 (/Users/stevenmcdonald/Documents/developer-servicev-1/build/classes/java/main started by stevenmcdonald in /Users/stevenmcdonald/Documents/developer-servicev-1)
2020-02-18 16:21:18.164  INFO 26216 --- [main] i.d.d.DeveloperServiceApplication        : No active profile set, falling back to default profiles: default
2020-02-18 16:21:18.720  INFO 26216 --- [main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data Reactive MongoDB repositories in DEFAULT mode.
2020-02-18 16:21:18.730  INFO 26216 --- [main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 7ms. Found 0 Reactive MongoDB repository interfaces.
2020-02-18 16:21:18.733  INFO 26216 --- [main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data MongoDB repositories in DEFAULT mode.
2020-02-18 16:21:18.733  INFO 26216 --- [main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 0ms. Found 0 MongoDB repository interfaces.
2020-02-18 16:21:19.111  INFO 26216 --- [main] org.mongodb.driver.cluster               : Cluster created with settings {hosts=[localhost:27017], mode=SINGLE, requiredClusterType=UNKNOWN, serverSelectionTimeout='30000 ms', maxWaitQueueSize=500}
2020-02-18 16:21:19.191  INFO 26216 --- [cluster-ClusterId{value='5e4c635f029a5221e3f3b2e0', description='null'}-localhost:27017] org.mongodb.driver.cluster               : Exception in monitor thread while connecting to server localhost:27017

com.mongodb.MongoSocketOpenException: Exception opening socket
	at com.mongodb.connection.netty.NettyStream$OpenChannelFutureListener.operationComplete(NettyStream.java:410) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.connection.netty.NettyStream$OpenChannelFutureListener.operationComplete(NettyStream.java:385) ~[mongodb-driver-core-3.11.2.jar:na]
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:577) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:570) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:549) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:490) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:615) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:608) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:117) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.fulfillConnectPromise(AbstractNioChannel.java:321) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:337) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:702) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_241]
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017
Caused by: java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method) ~[na:1.8.0_241]
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717) ~[na:1.8.0_241]
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:330) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:702) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493) ~[netty-transport-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.45.Final.jar:4.1.45.Final]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_241]

2020-02-18 16:21:19.362  INFO 26216 --- [main] o.s.b.a.e.web.EndpointLinksResolver      : Exposing 2 endpoint(s) beneath base path '/actuator'
2020-02-18 16:21:19.565  INFO 26216 --- [main] o.s.b.web.embedded.netty.NettyWebServer  : Netty started on port(s): 8080
2020-02-18 16:21:19.568  INFO 26216 --- [main] i.d.d.DeveloperServiceApplication        : Started DeveloperServiceApplication in 1.707 seconds (JVM running for 2.184)
2020-02-18 16:21:20.024  INFO 26216 --- [RMI TCP Connection(1)-10.0.0.155] org.mongodb.driver.cluster               : No server chosen by com.mongodb.async.client.ClientSessionHelper$1@4bad708f from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
2020-02-18 16:21:50.036  INFO 26216 --- [cluster-5e4c635f029a5221e3f3b2e0] org.mongodb.driver.cluster               : No server chosen by ReadPreferenceServerSelector{readPreference=primary} from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]}. Waiting for 30000 ms before timing out
2020-02-18 16:22:20.043  WARN 26216 --- [cluster-5e4c635f029a5221e3f3b2e0] o.s.b.a.m.MongoReactiveHealthIndicator   : Mongo health check failed

org.springframework.dao.DataAccessResourceFailureException: Timed out after 30000 ms while waiting for a server that matches ReadPreferenceServerSelector{readPreference=primary}. Client view of cluster state is {type=UNKNOWN, servers=[{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]; nested exception is com.mongodb.MongoTimeoutException: Timed out after 30000 ms while waiting for a server that matches ReadPreferenceServerSelector{readPreference=primary}. Client view of cluster state is {type=UNKNOWN, servers=[{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]
	at org.springframework.data.mongodb.core.MongoExceptionTranslator.translateExceptionIfPossible(MongoExceptionTranslator.java:90) ~[spring-data-mongodb-2.2.4.RELEASE.jar:2.2.4.RELEASE]
	at org.springframework.data.mongodb.core.ReactiveMongoTemplate.potentiallyConvertRuntimeException(ReactiveMongoTemplate.java:2777) ~[spring-data-mongodb-2.2.4.RELEASE.jar:2.2.4.RELEASE]
	at org.springframework.data.mongodb.core.ReactiveMongoTemplate.lambda$translateException$87(ReactiveMongoTemplate.java:2760) ~[spring-data-mongodb-2.2.4.RELEASE.jar:2.2.4.RELEASE]
	at reactor.core.publisher.Flux.lambda$onErrorMap$28(Flux.java:6430) ~[reactor-core-3.3.2.RELEASE.jar:3.3.2.RELEASE]
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:88) ~[reactor-core-3.3.2.RELEASE.jar:3.3.2.RELEASE]
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyInner.onError(MonoFlatMapMany.java:247) ~[reactor-core-3.3.2.RELEASE.jar:3.3.2.RELEASE]
	at com.mongodb.reactivestreams.client.internal.SingleResultObservableToPublisher$1.onError(SingleResultObservableToPublisher.java:50) ~[mongodb-driver-reactivestreams-1.12.0.jar:na]
	at com.mongodb.reactivestreams.client.internal.ObservableToPublisher$1.onError(ObservableToPublisher.java:71) ~[mongodb-driver-reactivestreams-1.12.0.jar:na]
	at com.mongodb.async.client.AbstractSubscription.onError(AbstractSubscription.java:129) ~[mongodb-driver-async-3.11.2.jar:na]
	at com.mongodb.async.client.SingleResultCallbackSubscription$1.onResult(SingleResultCallbackSubscription.java:42) ~[mongodb-driver-async-3.11.2.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.async.client.OperationExecutorImpl$1$1$1.onResult(OperationExecutorImpl.java:94) ~[mongodb-driver-async-3.11.2.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.operation.OperationHelper$ReferenceCountedReleasingWrappedCallback.onResult(OperationHelper.java:412) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.operation.CommandOperationHelper$8.call(CommandOperationHelper.java:413) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.operation.OperationHelper$AsyncCallableWithConnectionAndSourceCallback.onResult(OperationHelper.java:629) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.operation.OperationHelper$AsyncCallableWithConnectionAndSourceCallback.onResult(OperationHelper.java:619) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:49) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.binding.AsyncClusterBinding$1.onResult(AsyncClusterBinding.java:111) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.binding.AsyncClusterBinding$1.onResult(AsyncClusterBinding.java:107) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.internal.connection.BaseCluster$ServerSelectionRequest.onResult(BaseCluster.java:440) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.internal.connection.BaseCluster.handleServerSelectionRequest(BaseCluster.java:316) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.internal.connection.BaseCluster.access$800(BaseCluster.java:66) ~[mongodb-driver-core-3.11.2.jar:na]
	at com.mongodb.internal.connection.BaseCluster$WaitQueueHandler.run(BaseCluster.java:491) ~[mongodb-driver-core-3.11.2.jar:na]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_241]
Caused by: com.mongodb.MongoTimeoutException: Timed out after 30000 ms while waiting for a server that matches ReadPreferenceServerSelector{readPreference=primary}. Client view of cluster state is {type=UNKNOWN, servers=[{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:27017}, caused by {java.net.ConnectException: Connection refused}}]
	at com.mongodb.internal.connection.BaseCluster.createTimeoutException(BaseCluster.java:408) ~[mongodb-driver-core-3.11.2.jar:na]
	... 4 common frames omitted

2020-02-18 16:28:42.807  INFO 26280 --- [main] i.d.d.DeveloperServiceApplication        : Starting DeveloperServiceApplication on Stevens-MacBook-Pro.local with PID 26280 (/Users/stevenmcdonald/Documents/developer-servicev-1/build/classes/java/main started by stevenmcdonald in /Users/stevenmcdonald/Documents/developer-servicev-1)
2020-02-18 16:28:42.808  INFO 26280 --- [main] i.d.d.DeveloperServiceApplication        : No active profile set, falling back to default profiles: default
2020-02-18 16:28:43.620  INFO 26280 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8080 (http)
2020-02-18 16:28:43.626  INFO 26280 --- [main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2020-02-18 16:28:43.627  INFO 26280 --- [main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.30]
2020-02-18 16:28:43.682  INFO 26280 --- [main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2020-02-18 16:28:43.682  INFO 26280 --- [main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 848 ms
2020-02-18 16:28:43.929  INFO 26280 --- [main] o.s.s.concurrent.ThreadPoolTaskExecutor  : Initializing ExecutorService 'applicationTaskExecutor'
2020-02-18 16:28:44.130  INFO 26280 --- [main] o.s.b.a.e.web.EndpointLinksResolver      : Exposing 2 endpoint(s) beneath base path '/actuator'
2020-02-18 16:28:44.182  INFO 26280 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8080 (http) with context path ''
2020-02-18 16:28:44.186  INFO 26280 --- [main] i.d.d.DeveloperServiceApplication        : Started DeveloperServiceApplication in 1.625 seconds (JVM running for 2.408)
2020-02-18 16:28:44.548  INFO 26280 --- [RMI TCP Connection(2)-127.0.0.1] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring DispatcherServlet 'dispatcherServlet'
2020-02-18 16:28:44.548  INFO 26280 --- [RMI TCP Connection(2)-127.0.0.1] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2020-02-18 16:28:44.553  INFO 26280 --- [RMI TCP Connection(2)-127.0.0.1] o.s.web.servlet.DispatcherServlet        : Completed initialization in 5 ms
2020-02-18 16:29:36.929  INFO 26280 --- [SpringContextShutdownHook] o.s.s.concurrent.ThreadPoolTaskExecutor  : Shutting down ExecutorService 'applicationTaskExecutor'
2020-02-18 16:29:47.130  INFO 26287 --- [main] i.d.d.DeveloperServiceApplication        : Starting DeveloperServiceApplication on Stevens-MacBook-Pro.local with PID 26287 (/Users/stevenmcdonald/Documents/developer-servicev-1/build/classes/java/main started by stevenmcdonald in /Users/stevenmcdonald/Documents/developer-servicev-1)
2020-02-18 16:29:47.131  INFO 26287 --- [main] i.d.d.DeveloperServiceApplication        : No active profile set, falling back to default profiles: default
2020-02-18 16:29:47.820  INFO 26287 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8080 (http)
2020-02-18 16:29:47.830  INFO 26287 --- [main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2020-02-18 16:29:47.830  INFO 26287 --- [main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.30]
2020-02-18 16:29:47.881  INFO 26287 --- [main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2020-02-18 16:29:47.882  INFO 26287 --- [main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 730 ms
2020-02-18 16:29:48.098  INFO 26287 --- [main] o.s.s.concurrent.ThreadPoolTaskExecutor  : Initializing ExecutorService 'applicationTaskExecutor'
2020-02-18 16:29:48.223  INFO 26287 --- [main] o.s.b.a.e.web.EndpointLinksResolver      : Exposing 2 endpoint(s) beneath base path '/actuator'
2020-02-18 16:29:48.260  INFO 26287 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8080 (http) with context path ''
2020-02-18 16:29:48.263  INFO 26287 --- [main] i.d.d.DeveloperServiceApplication        : Started DeveloperServiceApplication in 1.405 seconds (JVM running for 1.936)
2020-02-18 16:29:48.605  INFO 26287 --- [RMI TCP Connection(2)-127.0.0.1] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring DispatcherServlet 'dispatcherServlet'
2020-02-18 16:29:48.606  INFO 26287 --- [RMI TCP Connection(2)-127.0.0.1] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2020-02-18 16:29:48.610  INFO 26287 --- [RMI TCP Connection(2)-127.0.0.1] o.s.web.servlet.DispatcherServlet        : Completed initialization in 4 ms
2020-02-18 16:30:43.359  INFO 26287 --- [SpringContextShutdownHook] o.s.s.concurrent.ThreadPoolTaskExecutor  : Shutting down ExecutorService 'applicationTaskExecutor'
2020-02-18 16:31:20.516  INFO 26306 --- [main] i.d.d.DeveloperServiceApplication        : Starting DeveloperServiceApplication on Stevens-MacBook-Pro.local with PID 26306 (/Users/stevenmcdonald/Documents/developer-servicev-1/build/classes/java/main started by stevenmcdonald in /Users/stevenmcdonald/Documents/developer-servicev-1)
2020-02-18 16:31:20.519  INFO 26306 --- [main] i.d.d.DeveloperServiceApplication        : No active profile set, falling back to default profiles: default
2020-02-18 16:31:21.471  INFO 26306 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8080 (http)
2020-02-18 16:31:21.479  INFO 26306 --- [main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2020-02-18 16:31:21.479  INFO 26306 --- [main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.30]
2020-02-18 16:31:21.541  INFO 26306 --- [main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2020-02-18 16:31:21.542  INFO 26306 --- [main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 990 ms
2020-02-18 16:31:21.891  INFO 26306 --- [main] o.s.s.concurrent.ThreadPoolTaskExecutor  : Initializing ExecutorService 'applicationTaskExecutor'
2020-02-18 16:31:22.084  INFO 26306 --- [main] o.s.b.a.e.web.EndpointLinksResolver      : Exposing 2 endpoint(s) beneath base path '/actuator'
2020-02-18 16:31:22.143  INFO 26306 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8080 (http) with context path ''
2020-02-18 16:31:22.147  INFO 26306 --- [main] i.d.d.DeveloperServiceApplication        : Started DeveloperServiceApplication in 2.027 seconds (JVM running for 2.971)
2020-02-18 16:31:22.518  INFO 26306 --- [RMI TCP Connection(2)-127.0.0.1] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring DispatcherServlet 'dispatcherServlet'
2020-02-18 16:31:22.519  INFO 26306 --- [RMI TCP Connection(2)-127.0.0.1] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2020-02-18 16:31:22.526  INFO 26306 --- [RMI TCP Connection(2)-127.0.0.1] o.s.web.servlet.DispatcherServlet        : Completed initialization in 6 ms
2020-02-18 16:32:03.870  INFO 26306 --- [SpringContextShutdownHook] o.s.s.concurrent.ThreadPoolTaskExecutor  : Shutting down ExecutorService 'applicationTaskExecutor'
2020-02-18 16:32:08.277  INFO 26315 --- [main] i.d.d.DeveloperServiceApplication        : Starting DeveloperServiceApplication on Stevens-MacBook-Pro.local with PID 26315 (/Users/stevenmcdonald/Documents/developer-servicev-1/build/classes/java/main started by stevenmcdonald in /Users/stevenmcdonald/Documents/developer-servicev-1)
2020-02-18 16:32:08.278  INFO 26315 --- [main] i.d.d.DeveloperServiceApplication        : No active profile set, falling back to default profiles: default
2020-02-18 16:32:09.016  INFO 26315 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8080 (http)
2020-02-18 16:32:09.022  INFO 26315 --- [main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2020-02-18 16:32:09.022  INFO 26315 --- [main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.30]
2020-02-18 16:32:09.073  INFO 26315 --- [main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2020-02-18 16:32:09.073  INFO 26315 --- [main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 774 ms
2020-02-18 16:32:09.309  INFO 26315 --- [main] o.s.s.concurrent.ThreadPoolTaskExecutor  : Initializing ExecutorService 'applicationTaskExecutor'
2020-02-18 16:32:09.447  INFO 26315 --- [main] o.s.b.a.e.web.EndpointLinksResolver      : Exposing 2 endpoint(s) beneath base path '/actuator'
2020-02-18 16:32:09.503  INFO 26315 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8080 (http) with context path ''
2020-02-18 16:32:09.507  INFO 26315 --- [main] i.d.d.DeveloperServiceApplication        : Started DeveloperServiceApplication in 1.484 seconds (JVM running for 1.954)
2020-02-18 16:32:09.677  INFO 26315 --- [RMI TCP Connection(2)-127.0.0.1] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring DispatcherServlet 'dispatcherServlet'
2020-02-18 16:32:09.678  INFO 26315 --- [RMI TCP Connection(2)-127.0.0.1] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2020-02-18 16:32:09.683  INFO 26315 --- [RMI TCP Connection(2)-127.0.0.1] o.s.web.servlet.DispatcherServlet        : Completed initialization in 4 ms
2020-02-18 16:33:25.340  INFO 26315 --- [SpringContextShutdownHook] o.s.s.concurrent.ThreadPoolTaskExecutor  : Shutting down ExecutorService 'applicationTaskExecutor'
